# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../pgbouncer/pgbouncer-sys_poll.h.patch
# Copyright (C) 2013 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

The standard is clearly defining that only <poll.h> needs to be included.

http://pubs.opengroup.org/onlinepubs/000095399/basedefs/poll.h.html

--- a/lib/m4/usual.m4	2013-08-09 15:16:56.716600440 +0200
+++ b/lib/m4/usual.m4	2013-08-09 15:17:07.488877797 +0200
@@ -165,7 +165,7 @@
 dnl  AC_USUAL_HEADER_CHECK:  Basic headers
 dnl
 AC_DEFUN([AC_USUAL_HEADER_CHECK], [
-AC_CHECK_HEADERS([sys/socket.h poll.h sys/poll.h sys/un.h])
+AC_CHECK_HEADERS([sys/socket.h poll.h sys/un.h])
 AC_CHECK_HEADERS([arpa/inet.h netinet/in.h netinet/tcp.h])
 AC_CHECK_HEADERS([sys/param.h sys/uio.h pwd.h grp.h])
 AC_CHECK_HEADERS([sys/wait.h sys/mman.h syslog.h netdb.h dlfcn.h])
--- a/lib/usual/socket.h	2013-08-09 15:10:58.444331854 +0200
+++ b/lib/usual/socket.h	2013-08-09 15:14:24.957124489 +0200
@@ -22,7 +22,6 @@
  * - win32: <winsock2.h>
  * - win32: <ws2tcpip.h>
  * - <sys/socket.h>
- * - <sys/poll.h>
  * - <sys/un.h>
  * - <netinet/in.h>
  * - <netinet/tcp.h>
@@ -50,9 +49,6 @@
 #ifdef HAVE_POLL_H
 #include <poll.h>
 #endif
-#ifdef HAVE_SYS_POLL_H
-#include <sys/poll.h>
-#endif
 #ifdef HAVE_SYS_UIO_H
 #include <sys/uio.h>
 #endif
